lms/djangoapps/learner_dashboard/api/v0/views.py

```diff
@@ -335,3 +338,47 @@ def get(self, request, program_uuid):
                 'credit_pathways': credit_pathways,
             }
         )
+
+
+class CourseRecommendationApiView(APIView):
+
+    """
+        **Example Request**
+
+        GET api/dashboard/v0/recommendation/courses/
+    """
+
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        """
+            Retrieves course recommendations details of a user in a specified course.
+        """
+        user_id = request.user.id
+        is_control, course_keys = get_personalized_course_recommendations(user_id)
+
+        # Emits an event to track student dashboard page visits.
+        segment.track(
+            user_id,
+            'edx.bi.student-dashboard-page.viewed',
+            {
+                'is_personalized_recommendation': not is_control,
+            }
+        )
+
+        if is_control:
+            return Response(status=400)
+        recommended_courses = []
+        for course_id in course_keys:
+            course_data = get_course_data(course_id)
+            if course_data:
+                recommended_courses.append({
+                    'course_key': course_data['key'],
+                    'title': str(course_data['title']),
+                    'logo_image_url': course_data['owners'][0]['logo_image_url'],
+                    'marketing_url': course_data['course_runs'][0]['marketing_url']
```

Comment:
Use `marketing_url` from the course data. e.g. `course_data.get('marketing_url')`